pkg/sbom/io/decode.go

```diff
@@ -211,12 +215,19 @@ func (m *Decoder) decodeLibrary(c *core.Component) (*ftypes.Package, error) {
 
 	pkg.Identifier.BOMRef = c.PkgID.BOMRef
 	pkg.Licenses = c.Licenses
-	if len(c.Files) > 0 {
-		pkg.Digest = c.Files[0].Hash
+
+	for _, f := range c.Files {
+		if f.Path != "" && pkg.FilePath == "" {
+			pkg.FilePath = f.Path
+		}
+		// An empty path represents a package digest
+		if f.Path == "" && len(f.Digests) > 0 {
```

Comment:
>Why can't we just add digest if it exists?

The package digest doesn't have a file path. I didn't want to have digests in several places. Any ideas?

Reply:
Got it!

Reply:
Perhaps I don't fully understand the reason for this condition.

We use filePath and calculate digest for [Applications](https://github.com/knqyf263/trivy/blob/7fdcf3499367bfbb5e011c90ce8fc0c5c31ca88f/pkg/fanal/types/const.go#L84-L90).
We also save both fields in SPDX format:
https://github.com/aquasecurity/trivy/blob/71da44f7e14705678ceaecc6f651e3a867dd6811/pkg/sbom/spdx/testdata/happy/with-files-in-relationships-bom.json#L58-L67

After decoding this File (after [adding digests support](https://github.com/knqyf263/trivy/blob/7fdcf3499367bfbb5e011c90ce8fc0c5c31ca88f/pkg/sbom/spdx/unmarshal.go#L178)) we will miss digest (because f.Path is not empty).

Correct me if i missed something.

Reply:
Now, we're not parsing digests in SPDX.
https://github.com/aquasecurity/trivy/blob/a14c913c85d80bcc52787878598465cdbf68e0a0/pkg/sbom/spdx/unmarshal.go#L177-L182

Reply:
ok, let's leave this logic and come back to this case after adding logic to handle checksums (if needed).